### PR TITLE
Changed http status code from StatusBadRequest to StatusForbidden

### LINKED ISSUE #2373: Error messages are not passed from virt-api to virtctl
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
It seems we don't pass error messages to virtctl from virt-api

I tried to run:

```./cluster/kubectl.sh create -f cluster/examples/vm-cirros.yaml```

spec.running is set to false so according to code run strategy should be Halted, when I execute

```./cluster/virtctl.sh restart vm-cirros```

I get:

```Error restarting VirtualMachine the server rejected our request for an unknown reason```

**What you expected to happen**:
I expect (https://github.com/kubevirt/kubevirt/pull/2311/files#diff-b2a2ac840fa7df47e840a82a10265a3bR238)

```runstrategy halted does not support manual restart requests```

**How to reproduce it (as minimally and precisely as possible)**:
```./cluster/kubectl.sh create -f cluster/examples/vm-cirros.yaml```
```./cluster/virtctl.sh restart vm-cirros```

**Environment**:
KubeVirt version (use `virtctl version`): 

Client Version: version.Info{GitVersion:"v0.18.0-29+06e6c55aea348c-dirty", GitCommit:"06e6c55aea348cf54fbf39fccafa08ca150a7c4e", GitTreeState:"dirty", BuildDate:"2019-06-12T08:27:23Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{GitVersion:"v0.18.0-29+06e6c55aea348c-dirty", GitCommit:"06e6c55aea348cf54fbf39fccafa08ca150a7c4e", GitTreeState:"dirty", BuildDate:"2019-06-12T08:27:23Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}

Kubernetes version (use `kubectl version`): 

Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-01T20:08:12Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-01T20:00:57Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}

### PR DESCRIPTION
**Pull request by vatsalparekh**
**What this PR does / why we need it**:
In virt-api subresources, Changed http status code from StatusBadRequest to StatusForbidden to present correct status code and errors
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2373 

**Special notes for your reviewer**:

**Release note**:

```release-note
Changed http status code from StatusBadRequest to StatusForbidden
```

### CODE REVIEW

File: pkg/virt-api/rest/subresource_test.go [hunk 1]
```diff
@@ -384,7 +384,7 @@ var _ = Describe("VirtualMachineInstance Subresources", func() {
 			app.RestartVMRequestHandler(request, response)
 
 			Expect(response.Error()).To(HaveOccurred())
-			Expect(response.StatusCode()).To(Equal(http.StatusBadRequest))
+			Expect(response.StatusCode()).To(Equal(http.StatusForbidden))
```
Comment (rmohr) [inline pkg/virt-api/rest/subresource_test.go]: We also want to check that `response.Error().Error()` really is the string which we expected.

Comment (vatsalparekh) [reply pkg/virt-api/rest/subresource_test.go]: Added.

**Pull request merged**